### PR TITLE
US4 Backend: Student Preferences

### PR DESCRIPTION
**Pull request by pharapeti**
## What is this change doing?

- Create api/student/preference_selections endpoint to get of the user's preferences
- Create api/student/preference_selections POST endpoint to create new preference_selection
- Create api/student/preference_selections/:id DELETE endpoint to allow student to delete preference_selection

## Why is it needed?

To allow the student to see their preferences
To allow the student to create new preferences

## Where should the reviewer start?

Routes, controllers

## How can these changes be manually tested?

See Postman

## What user story does this PR relate to? (include link)

US4

### CODE REVIEW

File: controllers/admin/projects/projects.controller.js [hunk 1]
```diff
@@ -29,10 +29,10 @@ exports.findOne = (req, res) => {
   })
 }
 
-// Create Project
+// Create Project associated to the admin creating it
 exports.createOne = (req, res) => {
   if (req.body.name && req.body.max_group_size) {
-    model.User.findOne({ external_user_id: req.session.external_user_id }).then(user => {
+    model.User.findOne({ where: { external_id: req.session.external_id }}).then(user => {
```
Comment (pharapeti) [inline controllers/admin/projects/projects.controller.js:35]: Fixed this bit

File: controllers/student/preference_selections/preference_selection.controller.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        res.status(500).send({ message: err.message || "Some error occurred while retrieving Preference Selections." });
+      })
+  })
+  .catch(err => {
+    res.status(500).send({ message: err.message || "Some error occurred while retrieving User." });
+  })
+}
+
+// Delete a single Preference Selection that is associated to the logged in user
+exports.deleteOne = (req, res) => {
```
Comment (DaGev) [inline controllers/student/preference_selections/preference_selection.controller.js:83]: There is an issue with destroying. It seems that it at no point do you grab a preference_selection to destroy. It looks like it should work by passing an id as a param, but it doesn't work regardless of how I pass it, as a param, in the body or the header. Am I missing something or is this an issue?

**Review by DaGev — COMMENTED** [review]
Left a comment on the destory function. Everything works besides that.

File: controllers/student/preference_selections/preference_selection.controller.js (hunk 2 above)
Comment (pharapeti) [reply controllers/student/preference_selections/preference_selection.controller.js:83]: That was intended. The `preference_selection` only really exists as a possible candidate of a `preference_category`. We can destroy it, but it doesn't make too much of a difference. In the real work, we would have a `findOrCreate` on the `preference category` to simplify the number of `preference_category` and `preference` combinations.

**Review by DaGev — APPROVED** [review]

**Pull request merged**